### PR TITLE
Add support for replace-existing-checks in api/ package

### PR DESCRIPTION
**Pull request by ShimmerGlass**

### CODE REVIEW

**Comment by fastest963** [comment]
Thanks @Aestek. LGTM

File: api/agent.go [hunk 1]
```diff
@@ -149,6 +149,8 @@ type AgentServiceRegistration struct {
 	Checks            AgentServiceChecks
 	Proxy             *AgentServiceConnectProxyConfig `json:",omitempty"`
 	Connect           *AgentServiceConnect            `json:",omitempty"`
+
+	ReplaceExistingChecks bool `json:"-"`
```
Comment (i0rek) [inline api/agent.go]: Why is it excluded from the json parsing and the other ones are not?

**Comment by i0rek** [comment]
Any updates?

**Comment by i0rek** [comment]
Fixed slightly different in #7136.

**Pull request closed without merging**